core/modules/catalog-next/store/category/actions.ts

```diff
@@ -0,0 +1,134 @@
+// import Vue from 'vue'
+import { ActionTree } from 'vuex'
+import * as types from './mutation-types'
+import RootState from '@vue-storefront/core/types/RootState'
+import CategoryState from './CategoryState'
+import { quickSearchByQuery } from '@vue-storefront/core/lib/search'
+import { buildFilterProductsQuery } from '@vue-storefront/core/helpers'
+import { router } from '@vue-storefront/core/app'
+import FilterVariant from '../../types/FilterVariant'
+import { CategoryService } from '@vue-storefront/core/data-resolver'
+import { changeFilterQuery } from '../../helpers/filterHelpers'
+import { products } from 'config'
+import { configureProductAsync } from '@vue-storefront/core/modules/catalog/helpers'
+import { DataResolver } from 'core/data-resolver/types/DataResolver';
+import { Category } from '../../types/Category';
+import { _prepareCategoryPathIds } from '../../helpers/categoryHelpers';
+import chunk from 'lodash-es/chunk'
+
+const actions: ActionTree<CategoryState, RootState> = {
+  async loadCategoryProducts ({ commit, getters, dispatch, rootState }, { route, category } = {}) {
```

Comment:
imho products should be cached by default. No additional work should be required to do this